### ISSUE TITLE
Undefined property PhpParser\Node\Param::$flags

### ISSUE THREAD

**Issue by javaDeveloperKid** [rectorphp/rector#3640]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | ^0.7.41 or 0.8.x-dev@47fc574 (docker image)              |
| Installed as   | composer dependency, composer bin dependency, docker image                      |

```
Notice: Undefined property: PhpParser\Node\Param::$flags in /usr/src/app/api/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinter/Standard.php on line 21
```
and
```
 [ERROR] Could not process "src/Services/Product/TaxService.php" file, due to:  
         "Argument 1 passed to PhpParser\PrettyPrinterAbstract::pModifiers()    
         must be of the type int, null given, called in                         
         /usr/src/app/api/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinter/St
         andard.php on line 21".
```

My rector.yaml:
```yaml
services:
    Rector\Php74\Rector\Property\TypedPropertyRector:

parameters:
    auto_import_names: true
    import_short_classes: false
    autoload_paths:
        - 'vendor/autoload.php'
    paths:
        - 'src'
```

What is going on? Notice and error message happen tens of times on a single rector run.
The same happens when installing rector as vendor-bin dependency (using bamarni/composer-bin-plugin).
Third option I tried was docker. It works with --dry run (`docker run -it --rm -v $(pwd):/project rector/rector:latest process --config /project/rector.yaml --dry-run`, P.S. yes, config paths were updated to match /project) but without dry run an error occurs:
```
Fatal error: Declaration of Symfony\Component\Cache\Adapter\ArrayAdapter::clear() must be compatible with Symfony\Component\Cache\Adapter\AdapterInterface::clear(string $prefix = '') in /project/vendor/symfony/cache/Adapter/ArrayAdapter.php on line 76
``` 
My first hint to problem with ArrayAdapter fatal error is: My project uses SF ^4.4.*, but rector from latest image uses ^5.0.6. It may conflict when I add vendor/autoload.php to the rector.yaml but shouldn't rector handle that? If not then there should be a statement in docs not to use rector version that has dependencies not matching developer's project dependencies.

**Comment by TomasVotruba** [comment]
That's probably a bug in php-parser, I'll fix it there.

> Fatal error: Declaration of...

This always means you need a prefixed Rector, that has different classes than your application.

**Comment by javaDeveloperKid** [comment]
Thank you.
For now I've installed 0.5.23. I will check out prefixed Rector in my spare time.

**Comment by javaDeveloperKid** [comment]
Now there is another problem. Rector adds types for overwritten properties that do not have type in parent class.what causes fatal error. E.g. `Symfony\Component\Console\Command\Command::$defaultName` which I overwrite in my personal command. Should I open a new issue for that?

**Comment by TomasVotruba** [comment]
>  Should I open a new issue for that?

If the problem remains on `dev-master`, then yes

**Comment by javaDeveloperKid** [comment]
NIce https://getrector.org/demo/a4ac13ad-8922-460d-a250-205000cd6f65#result

**Comment by TomasVotruba** [comment]
It's the flags issues again.

**Comment by javaDeveloperKid** [comment]
Yes. It looks like I have to wait for this to be fixed, and then I will be able to check if this another issue remains on dev-master.

**Comment by javaDeveloperKid** [comment]
I've run demo again and it works - it does not add type to the overwritten variable.

I've also run docker image again (the same image as in issue description, did not update latest tag after your changes 8 hours ago) and this problem with **Fatal error: Declaration of...** did not occur. That's awkward. nikic-php-parser is at 4.5.0 and symfony/* packages are at 5.1.2 in rector's composer.lock (checked from /rector inside container with `composer show`).

**Comment by TomasVotruba** [comment]
Solved by locking down php-parser to 4.5, until phpstan upgrades to 4.6.

This is released and tagged in Rector
